### ISSUE TITLE
widewine not working on  rp3b+

### ISSUE THREAD

**Issue by DarthRaver** [emilsvennesson/script.module.inputstreamhelper#411]
rpi3b+
Kodi version 18.7
inputstreamhelper version = 0.5.1
inputstreamadaptive = 2.4.4

Widewine (Crackle) was a working for a while, but some month ago it stopped.
Now i tried to search for a fix and i found that comment:

> 
> ### If you are affected by this issue now, follow these steps:
> 
>     * Go to **Add-ons** » **Program add-ons**
> 
>     * Start **InputStream Helper**
> 
>     * Select and click on **Restore Widevine CDM library...**
> 
>     * Pick any version → stay clear of version **4.10.1679.0 (13310.91.0)**
> 
> 
> **Update:** A newer release of InputStream Helper prevents the issue from happening onward.

idk why but i dont have backups, other than 4.10.1679.0.
and on all of my attempts to reinstall, it always installs 4.10.1679.0.

so what do i have to do to get a working version of inputstreamhelper and widewine?

Thanks.

**Comment by dagwieers** [comment]
@DarthRaver Since you are running ISH v0.5.1 you are not impacted by the issue you found.

Why do you think your issue with Crackle is related to Widevine?

**Comment by DarthRaver** [comment]
The logs and my search magic led me to the conclusion that it _could_ be related.
If not, please point me in the right direction, as my understanding of that stuff is pretty limited.
So it is ok to have version 4.10.1679.0 installed?
Here is a part of the log, trying to play an episode on crackle.
```
2020-12-06 09:11:35.808 T:1263526112   ERROR: /home/pi/.kodi/addons/script.module.urllib3/lib/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
                                              InsecureRequestWarning,
2020-12-06 09:11:36.072 T:1263526112   ERROR: [script.module.inputstreamhelper] [u'ldd', u'/home/pi/.kodi/cdm/libwidevinecdm.so'] cmd failed.
2020-12-06 09:11:36.076 T:1263526112   ERROR: [script.module.inputstreamhelper] Failed to check for missing Widevine libraries.
2020-12-06 09:11:36.078 T:1263526112   ERROR: EXCEPTION: Invalid setting type
2020-12-06 09:11:36.081 T:1263526112  NOTICE: [script.module.inputstreamhelper] Widevine update check was made on 2020-12-06 01:23
2020-12-06 09:11:36.462 T:1900340144  NOTICE: VideoPlayer::OpenFile: plugin://plugin.video.crackle/?id=2417092&mode=103&type=tvshows
2020-12-06 09:11:36.478 T:1322774752  NOTICE: Creating InputStream
2020-12-06 09:11:36.909 T:1322774752   ERROR: AddOnLog: InputStream Adaptive: Unable to load widevine shared library (/home/pi/.kodi/cdm/libwidevinecdm.so)
2020-12-06 09:11:36.912 T:1322774752   ERROR: AddOnLog: InputStream Adaptive: OpenDRMSystem failed
2020-12-06 09:11:36.922 T:1322774752   ERROR: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.crackle/?id=2417092&mode=103&type=tvshows]
2020-12-06 09:11:36.929 T:1322774752  NOTICE: CVideoPlayer::OnExit()
2020-12-06 09:11:37.003 T:1900340144  NOTICE: CVideoPlayer::CloseFile()
2020-12-06 09:11:37.018 T:1900340144  NOTICE: VideoPlayer: waiting for threads to exit
2020-12-06 09:11:37.020 T:1900340144  NOTICE: VideoPlayer: finished waiting
```

**Comment by dagwieers** [comment]
Can you remove this file _/home/pi/.kodi/cdm/libwidevinecdm.so_ ? And then try again?

**Comment by DarthRaver** [comment]
I tried that before, but was hoping for a different outcome.
It still dont play the file and it is again version 4.10.1679.0.
Still dont know if that version is the problem?

I have a few other warnings and errors in the log, but idk if they are related in any way.
```
2020-12-07 17:02:56.011 T:1899770800 WARNING: CSettingsManager: missing version attribute
2020-12-07 17:03:01.274 T:1899770800   ERROR: DBus error: org.freedesktop.DBus.Error.ServiceUnknown - The name org.freedesktop.UPower was not provided by any .service files
2020-12-07 17:03:02.368 T:1899770800 WARNING: COMXImageFile::GetCodingType progressive images not supported by decoder
2020-12-07 17:03:09.151 T:1899770800 WARNING: JSONRPC: Could not parse type "Setting.Details.SettingList"
2020-12-07 17:03:10.131 T:1899770800   ERROR: JSONRPC Server: Failed to connect to sdpd
2020-12-07 17:03:10.281 T:1565311200   ERROR: CVideoPlayer::OpenInputStream - error opening [/var/run/lirc/lircd]
2020-12-07 17:03:31.828 T:1752596704 WARNING: CreateLoader - unsupported protocol(thumb) in thumb://Current/
```
tried playing file. when asked, i installed widevine again, and after that i tried to play the file again.
```
2020-12-07 17:04:04.395 T:1681907936   ERROR: /home/pi/.kodi/addons/script.module.urllib3/lib/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
                                              InsecureRequestWarning,
2020-12-07 17:04:35.302 T:1427628256   ERROR: Previous line repeats 3 times.
2020-12-07 17:04:35.305 T:1427628256 WARNING: [script.module.inputstreamhelper] Widevine CDM is not installed.
2020-12-07 17:04:44.410 T:1427628256  NOTICE: [script.module.inputstreamhelper] Recursively create directory '/home/pi/.kodi/userdata/addon_data/script.module.inputstreamhelper/temp/'.
2020-12-07 17:04:44.780 T:1427628256 WARNING: [script.module.inputstreamhelper] Widevine CDM is not installed.
2020-12-07 17:04:45.289 T:1427628256  NOTICE: [script.module.inputstreamhelper] Recursively create directory '/home/pi/.kodi/userdata/addon_data/script.module.inputstreamhelper/temp/'.
2020-12-07 17:05:30.861 T:1899770800  NOTICE: Samba is idle. Closing the remaining connections
2020-12-07 17:11:19.161 T:1427628256  NOTICE: [script.module.inputstreamhelper] Recursively create directory '/home/pi/.kodi/cdm/'.
2020-12-07 17:11:19.984 T:1427628256  NOTICE: [script.module.inputstreamhelper] Hardlink file '/home/pi/.kodi/userdata/addon_data/script.module.inputstreamhelper/backup/13421.99.0/config.json' to '/home/pi/.kodi/cdm/config.json'.
2020-12-07 17:11:19.999 T:1427628256  NOTICE: [script.module.inputstreamhelper] Hardlink file '/home/pi/.kodi/userdata/addon_data/script.module.inputstreamhelper/backup/13421.99.0/recovery.json' to '/home/pi/.kodi/cdm/recovery.json'.
2020-12-07 17:11:20.013 T:1427628256  NOTICE: [script.module.inputstreamhelper] Hardlink file '/home/pi/.kodi/userdata/addon_data/script.module.inputstreamhelper/backup/13421.99.0/libwidevinecdm.so' to '/home/pi/.kodi/cdm/libwidevinecdm.so'.
2020-12-07 17:11:20.732 T:1427628256   ERROR: [script.module.inputstreamhelper] [u'ldd', u'/home/pi/.kodi/cdm/libwidevinecdm.so'] cmd failed.
2020-12-07 17:11:20.764 T:1427628256   ERROR: [script.module.inputstreamhelper] Failed to check for missing Widevine libraries.
2020-12-07 17:11:20.841 T:1427628256   ERROR: EXCEPTION: Invalid setting type
2020-12-07 17:11:20.845 T:1427628256  NOTICE: [script.module.inputstreamhelper] Widevine update check was made on 2020-12-06 01:23
2020-12-07 17:11:22.025 T:1899770800  NOTICE: VideoPlayer::OpenFile: plugin://plugin.video.crackle/?id=2511163&mode=103&type=tvshows
2020-12-07 17:11:22.043 T:1556918496  NOTICE: Creating InputStream
2020-12-07 17:11:24.151 T:1556918496   ERROR: AddOnLog: InputStream Adaptive: Unable to load widevine shared library (/home/pi/.kodi/cdm/libwidevinecdm.so)
2020-12-07 17:11:24.154 T:1556918496   ERROR: AddOnLog: InputStream Adaptive: OpenDRMSystem failed
2020-12-07 17:11:24.201 T:1556918496   ERROR: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.crackle/?id=2511163&mode=103&type=tvshows]
2020-12-07 17:11:24.204 T:1556918496  NOTICE: CVideoPlayer::OnExit()
2020-12-07 17:11:24.284 T:1899770800  NOTICE: CVideoPlayer::CloseFile()
2020-12-07 17:11:24.288 T:1899770800  NOTICE: VideoPlayer: waiting for threads to exit
2020-12-07 17:11:24.291 T:1899770800  NOTICE: VideoPlayer: finished waiting
2020-12-07 17:11:36.702 T:1752596704   ERROR: /home/pi/.kodi/addons/script.module.urllib3/lib/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
                                              InsecureRequestWarning,
2020-12-07 17:11:36.973 T:1752596704   ERROR: [script.module.inputstreamhelper] [u'ldd', u'/home/pi/.kodi/cdm/libwidevinecdm.so'] cmd failed.
2020-12-07 17:11:36.982 T:1752596704   ERROR: [script.module.inputstreamhelper] Failed to check for missing Widevine libraries.
2020-12-07 17:11:36.984 T:1752596704   ERROR: EXCEPTION: Invalid setting type
2020-12-07 17:11:36.987 T:1752596704  NOTICE: [script.module.inputstreamhelper] Widevine update check was made on 2020-12-07 17:11
2020-12-07 17:11:37.189 T:1899770800  NOTICE: VideoPlayer::OpenFile: plugin://plugin.video.crackle/?id=2511163&mode=103&type=tvshows
2020-12-07 17:11:37.206 T:1419235552  NOTICE: Creating InputStream
2020-12-07 17:11:37.344 T:1419235552   ERROR: AddOnLog: InputStream Adaptive: Unable to load widevine shared library (/home/pi/.kodi/cdm/libwidevinecdm.so)
2020-12-07 17:11:37.352 T:1419235552   ERROR: AddOnLog: InputStream Adaptive: OpenDRMSystem failed
2020-12-07 17:11:37.363 T:1419235552   ERROR: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.crackle/?id=2511163&mode=103&type=tvshows]
2020-12-07 17:11:37.385 T:1419235552  NOTICE: CVideoPlayer::OnExit()
2020-12-07 17:11:37.409 T:1899770800  NOTICE: CVideoPlayer::CloseFile()
2020-12-07 17:11:37.412 T:1899770800  NOTICE: VideoPlayer: waiting for threads to exit
2020-12-07 17:11:37.415 T:1899770800  NOTICE: VideoPlayer: finished waiting
2020-12-07 17:11:46.339 T:1752596704   ERROR: /home/pi/.kodi/addons/script.module.urllib3/lib/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
                                              InsecureRequestWarning,
2020-12-07 17:11:46.614 T:1752596704   ERROR: [script.module.inputstreamhelper] [u'ldd', u'/home/pi/.kodi/cdm/libwidevinecdm.so'] cmd failed.
2020-12-07 17:11:46.617 T:1752596704   ERROR: [script.module.inputstreamhelper] Failed to check for missing Widevine libraries.
2020-12-07 17:11:46.619 T:1752596704   ERROR: EXCEPTION: Invalid setting type
2020-12-07 17:11:46.622 T:1752596704  NOTICE: [script.module.inputstreamhelper] Widevine update check was made on 2020-12-07 17:11
2020-12-07 17:11:47.015 T:1899770800  NOTICE: VideoPlayer::OpenFile: plugin://plugin.video.crackle/?id=2511163&mode=103&type=tvshows
2020-12-07 17:11:47.066 T:1419235552  NOTICE: Creating InputStream
2020-12-07 17:11:47.175 T:1419235552   ERROR: AddOnLog: InputStream Adaptive: Unable to load widevine shared library (/home/pi/.kodi/cdm/libwidevinecdm.so)
2020-12-07 17:11:47.178 T:1419235552   ERROR: AddOnLog: InputStream Adaptive: OpenDRMSystem failed
2020-12-07 17:11:47.234 T:1419235552   ERROR: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.crackle/?id=2511163&mode=103&type=tvshows]
2020-12-07 17:11:47.236 T:1419235552  NOTICE: CVideoPlayer::OnExit()
2020-12-07 17:11:47.265 T:1899770800  NOTICE: CVideoPlayer::CloseFile()
2020-12-07 17:11:47.288 T:1899770800  NOTICE: VideoPlayer: waiting for threads to exit
2020-12-07 17:11:47.291 T:1899770800  NOTICE: VideoPlayer: finished waiting
2020-12-07 17:12:07.525 T:1752596704 WARNING: CreateLoader - unsupported protocol(thumb) in thumb://Current/
2020-12-07 17:12:18.198 T:1427628256   ERROR: EXCEPTION: Invalid setting type
2020-12-07 17:12:18.724 T:1427628256   ERROR: Previous line repeats 3 times.
2020-12-07 17:12:18.736 T:1427628256  NOTICE: [script.module.inputstreamhelper] 
                                            Kodi Version 18.7 läuft auf einem Linux System mit arm Architektur.
                                            
                                            InputStream Helper ist in Version 0.5.1 
                                            InputStream Adaptive ist in Version 2.4.4 
                                            
                                            Widevine CDM ist in Version 4.10.1679.0 und wurde installiert am 2020-12-07 17:11
                                            Es wurde extrahiert aus Chrome OS Image MICKEY mit Version 13421.99.0
                                            Zuletzt am 2020-12-07 17:11 auf Aktualisierungen geprüft
                                            Es ist installiert in /home/pi/.kodi/cdm/
                                            
                                            Bitte melden Sie Fehler hier: https://git.io/JfKJb
```
Thanks for your help.

**Comment by horstle** [comment]
It looks like the correct Widevine is installed, but the check for missing libs fails, which might mean `ldd` is not installed. So the problem could be some missing libraries like libnss3 or libnspr4.

**Comment by dagwieers** [comment]
@horstle If ldd fails (e.g. because it is not installed), maybe we should not assume there is a problem. It might as well work?

**Comment by horstle** [comment]
We already do. If that command fails, it gets logged, but continues anyway. That's not what is causing @DarthRaver's problem.

**Comment by DarthRaver** [comment]
Dont know if it helps
```
ldd /home/pi/.kodi/cdm/libwidevinecdm.so
das Programm ist nicht dynamisch gelinkt
(translated: program is not dynamicly linked) = not a dynamic executable?
```
libnss3,so and libnspr4.so also seems to be installed.

**Comment by horstle** [comment]
Ok, so ldd is installed and probably all necessary libs are there.

I just reinstalled Widevine on my RPi3 (Raspbian) and it works just fine. Could you run
`file /home/pi/.kodi/cdm/libwidevinecdm.so` and post the output here?
Also, the libwidevinecdm.so extracted from MICKEY version 13421.99.0 should have a sha1 checksum of `d5ee489d5c409ff6cb423f318a23dba71ff2f2b0`. Could you compare that by running
`sha1sum /home/pi/.kodi/cdm/libwidevinecdm.so`? I'm pretty sure that'll be different, but just to confirm.

And finally a full debug log could help, also because it could help to know which OS you are using.

**Comment by DarthRaver** [comment]
My OS is or was "Raspbian", now "Raspberry Pi OS", with berryboot as bootloader.

file /home/pi/.kodi/cdm/libwidevinecdm.so
```
/home/pi/.kodi/cdm/libwidevinecdm.so: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, BuildID[sha1]=0518f32c249a91d353237a2712aa9264ec8a8f6a, stripped
```
sha1sum /home/pi/.kodi/cdm/libwidevinecdm.so
```
d5ee489d5c409ff6cb423f318a23dba71ff2f2b0  /home/pi/.kodi/cdm/libwidevinecdm.so
```
Dont know from what you want a full debug log. Raspbian or kodi, or from the add-on?

**Comment by horstle** [comment]
I meant a debug log of kodi.

Ok, so you definitely have the correct libwidevinecdm.so at the correct place. Since the main point of inputstreamhelper is to ensure just that, the problem does not lie in this addon.

**Comment by matthuisman** [comment]
i solved the same error (Unable to load widevine shared library (/home/pi/.kodi/cdm/libwidevinecdm.so)) on RetroPie with

sudo apt-get install libnss3

But you said you had that installed?

**Comment by dagwieers** [comment]
> Dont know if it helps
> 
> ```
> ldd /home/pi/.kodi/cdm/libwidevinecdm.so
> das Programm ist nicht dynamisch gelinkt
> (translated: program is not dynamicly linked) = not a dynamic executable?
> ```
> 
> libnss3,so and libnspr4.so also seems to be installed.

This output is not correct, you should get something like this:
```
[dag@moria ~]$ ssh kodi01
##############################################
#                 LibreELEC                  #
#            https://libreelec.tv            #
##############################################

LibreELEC (official): 9.2.6 (RPi2.arm)
kodi01:~ # ls -la .kodi/cdm/libwidevinecdm.so 
-rwxr--r--    2 root     root       7297324 Dec  4 20:36 .kodi/cdm/libwidevinecdm.so

kodi01:~ # md5sum .kodi/cdm/libwidevinecdm.so
3144866613a7f691b8cb07df6bbc4d80  .kodi/cdm/libwidevinecdm.so

kodi01:~ # ldd .kodi/cdm/libwidevinecdm.so
	linux-vdso.so.1 (0x7eef8000)
	/usr/lib/libarmmem-v7l.so (0x76812000)
	libpthread.so.0 => /usr/lib/libpthread.so.0 (0x767e9000)
	libm.so.6 => /usr/lib/libm.so.6 (0x7677f000)
	libdl.so.2 => /usr/lib/libdl.so.2 (0x7676c000)
	librt.so.1 => /usr/lib/librt.so.1 (0x76755000)
	libnss3.so => /usr/lib/libnss3.so (0x76660000)
	libnssutil3.so => /usr/lib/libnssutil3.so (0x7662e000)
	libnspr4.so => /usr/lib/libnspr4.so (0x765fa000)
	libc.so.6 => /usr/lib/libc.so.6 (0x764be000)
	/usr/lib/ld-linux-armhf.so.3 (0x76f70000)
	libplc4.so => /usr/lib/libplc4.so (0x76f95000)
	libplds4.so => /usr/lib/libplds4.so (0x76f91000)

kodi01:~ # ls -la .kodi/cdm/config.json 
-rw-r--r--    2 root     root           608 Dec  4 20:36 .kodi/cdm/config.json

kodi01:~ # cat .kodi/cdm/config.json 
{"sku": "", "sha1": "e75223cb539bdd07254ba2afccc3a62e2ed81dbd", "name": "ASUS Chromebit CS10", "photourl": "", "url": "https://dl.google.com/dl/edgedl/chromeos/recovery/chromeos_13421.99.0_veyron-mickey_recovery_stable-channel_mickey-mp-v2.bin.zip", "hwid": "MICKEY", "hwids": [], "zipfilesize": 626756094, "version": "13421.99.0", "filesize": 2320612352, "file": "chromeos_13421.99.0_veyron-mickey_recovery_stable-channel_mickey-mp-v2.bin", "md5": "3a3ba03e1745d03638068a2263b67fd6", "hwidmatch": "^MICKEY .*", "desc": "", "model": "ASUS Chromebit CS10", "channel": "stable-channel", "manufacturer": "Asus"}
```

This is also on a RPi 3B running LibreELEC.

**Comment by DarthRaver** [comment]
I get results with ldd, after i copied libwidevinecdm.so to my Downloads folder and tried ldd on that file.
My kodi folder is hardlinked(?) to my USB hdd, and berryboot also links libs to some place.
So i guess there is "only" a problem with permissions?
```
~ $ ls -la .kodi/cdm/libwidevinecdm.so
-rwxr--r-- 2 pi pi 7297324 Dez  7 17:11 .kodi/cdm/libwidevinecdm.so

~ $ sha1sum .kodi/cdm/libwidevinecdm.so
d5ee489d5c409ff6cb423f318a23dba71ff2f2b0  .kodi/cdm/libwidevinecdm.so

~ $ ldd .kodi/cdm/libwidevinecdm.so
das Programm ist nicht dynamisch gelinkt
#(translated: program is not dynamicly linked) = not a dynamic executable?

~ $ ls -la .kodi/cdm/config.json
-rw-r--r-- 2 pi pi 608 Dez  7 17:11 .kodi/cdm/config.json

~ $ cat .kodi/cdm/config.json
{"sku": "", "sha1": "e75223cb539bdd07254ba2afccc3a62e2ed81dbd", "name": "ASUS Chromebit CS10", "photourl": "", "url": "https://dl.google.com/dl/edgedl/chromeos/recovery/chromeos_13421.99.0_veyron-mickey_recovery_stable-channel_mickey-mp-v2.bin.zip", "hwid": "MICKEY", "hwids": [], "zipfilesize": 626756094, "version": "13421.99.0", "filesize": 2320612352, "file": "chromeos_13421.99.0_veyron-mickey_recovery_stable-channel_mickey-mp-v2.bin", "md5": "3a3ba03e1745d03638068a2263b67fd6", "hwidmatch": "^MICKEY .*", "desc": "", "model": "ASUS Chromebit CS10", "channel": "stable-channel", "manufacturer": "Asus"}

ldd pointing to a copy of libwidevinecdm.so in /home/pi/Downloads
~/Downloads $ ldd libwidevinecdm.so
linux-vdso.so.1 (0x7ee99000)
/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so => /usr/lib/arm-linux-gnueabihf/libarmmem-v7l.so (0x76799000)
libpthread.so.0 => /lib/arm-linux-gnueabihf/libpthread.so.0 (0x7675a000)
libm.so.6 => /lib/arm-linux-gnueabihf/libm.so.6 (0x766d8000)
libdl.so.2 => /lib/arm-linux-gnueabihf/libdl.so.2 (0x766c5000)
librt.so.1 => /lib/arm-linux-gnueabihf/librt.so.1 (0x766ae000)
libnss3.so => /usr/lib/arm-linux-gnueabihf/libnss3.so (0x76592000)
libnssutil3.so => /usr/lib/arm-linux-gnueabihf/libnssutil3.so (0x7655f000)
libnspr4.so => /usr/lib/arm-linux-gnueabihf/libnspr4.so (0x7651f000)
libc.so.6 => /lib/arm-linux-gnueabihf/libc.so.6 (0x763d1000)
/lib/ld-linux-armhf.so.3 (0x76ef7000)
libplc4.so => /usr/lib/arm-linux-gnueabihf/libplc4.so (0x763bd000)
libplds4.so => /usr/lib/arm-linux-gnueabihf/libplds4.so (0x763aa000)
```

**Comment by Diniboy1123** [comment]
I am on LibreElec 9.2.4 atm, just updated the CDM using this addon. Since then I am experiencing the same issue, for some reason ldd cannot be called... I am thinking maybe the script cannot see it in the PATH or the permissions are wrong.

I use a Raspberry Pi 2.

**Comment by horstle** [comment]
@Diniboy1123 What you're experiencing is most probably what @DarthRaver thought was the problem at the beginning. I don't yet know why, but currently inputstreamhelper downloads the ChromeOS image for EXPRESSO, which is a x86 Chromebook, so the libwidevinecdm.so is incompatible. You can simply restore an older version of Widevine as mentioned in the first post.

**Comment by horstle** [comment]
Ok, I know why now. We didn't have a release since @mediaminister updated the ARM hwids in #401 .
We should do a release ASAP.

**Comment by mediaminister** [comment]
Release is done.

**Comment by sorin-pop** [comment]
Hi all,

I have the same issue here. A couple of days ago I got the usual "update Widevine" popup and I was foolish enough to say yes. Since then, videos are not starting.

I would have a few questions:

1. So if I understand correctly, this should be simply fixed after I update the INputStream Helper addon from my current 0.5.1 to 0.5.2?
2. Actually, when I choose the Update button, it does not show me version 0.5.2 as available yet.
3. How can I manually install 0.5.2?
4. I also tried the workaround of opening Inputstream Helper and using the Restore Widevine feature. However, when I open Inputstream Helper, its UI is completely broken, nothing is written out, just an empty window shown. Anybody experienced this?

Thank you.
